File: Examples/test-suite/perl5/constructor_copy_runme.pl
use strict;
use warnings;
use Test::More tests => 6;
BEGIN { use_ok('constructor_copy') }
require_ok('constructor_copy');

my $f1 = constructor_copy::Foo1->new(3);
my $f11 = constructor_copy::Foo1->new($f1);

is($f1->{x}, $f11->{x});


my $f8 = constructor_copy::Foo8->new();
is(eval { constructor_copy::Foo81->new($f8); 1 }, undef);


my $bi = constructor_copy::Bari->new(5);
my $bc = constructor_copy::Bari->new($bi);
is($bi->{x}, $bc->{x});
    
my $bd = constructor_copy::Bard->new(5);
is(eval { constructor_copy::Bard->new($bd); 1 }, undef);

File: Examples/test-suite/lua/constructor_copy_runme.lua
local v=require("constructor_copy")

local f1 = v.Foo1(3)
assert(f1 ~= nil)
local f11 = v.Foo1(f1)
assert(f11 ~= nil)
assert(f1.x == f11.x)

local f8 = v.Foo8()
assert(f8 ~= nil)
assert(not pcall(function() Foo8(f8) end))

local bi = v.Bari(5)
assert(bi ~= nil)
local bc = v.Bari(bi)
assert(bc ~= nil)
assert(bi.x == bc.x)

local bd = v.Bard(5)
assert(bd ~= nil)
assert(not pcall(function() Bard(bd) end))
